assert/forward_assertions.go

```diff
@@ -13,4 +13,9 @@ func New(t TestingT) *Assertions {
 	}
 }
 
+// New makes a new Assertions object for the subtest TestingT.
+func (a *Assertions) New(t TestingT) *Assertions {
```

Comment:
This seems confusing to me.

how about:

```go
func (a *Assertions) Run(name string, fn func(*Assertions)) {
 	a.t.Run(name, func(t *testing.T) {
 	 	a := New(t)
 	 	fn(a)
 	})
}
```

Reply:
Hi @ernesto-jimenez, thank you for the great idea! It looks compelling and I can easily imagine myself writing like:

```go
func Test_run(t *testing.T) {
	assert := assert.New(t)

	assert.Run("subtest", func(assert *assert.Assertions) { // Wait, package `assert` is shadowed
		assert.True(true)
	})
}
```

Oh wait, `*assert.Assertions` will be shadowed if I keep declaring `assert := assert.New(t)`. Also, `a.t` is a `TestingT` instead of `*testing.T` which doesn't have `Run()` method.